src/clj/ataru/virkailija/virkailija_routes.clj

```diff
@@ -200,54 +200,46 @@
-
-                                                    (some? email)
-                                                    (access-controlled-application/get-application-list-by-email email session organization-service)
-
-                                                    (some? name)
-                                                    (access-controlled-application/get-application-list-by-name name session organization-service))
-                                                  :applications)
-                                 persons      (->> (person-service/get-persons person-service (distinct (keep :person-oid applications)))
-                                                   (reduce (fn [res person]
-                                                             (assoc res (:oidHenkilo person) person))
-                                                           {}))]
-                             (response/ok {:applications (for [application applications
-                                                               :let [person      (get persons (:person-oid application))
-                                                                     yksiloity   (or (-> person :yksiloity)
-                                                                                     (-> person :yksiloityVTJ))
-                                                                     person-info (if yksiloity
-                                                                                   {:preferred-name (:kutsumanimi person)
-                                                                                    :last-name      (:sukunimi person)}
-                                                                                   (select-keys application [:preferred-name :last-name]))]]
-                                                           (-> application
-                                                               (assoc :person person-info)
-                                                               (dissoc :person-oid :preferred-name :last-name)))})))
+                   (api/GET "/list" {session :session}
+                     :query-params [{formKey :- s/Str nil}
+                                    {hakukohdeOid :- s/Str nil}
+                                    {hakuOid :- s/Str nil}
+                                    {ssn :- s/Str nil}
+                                    {dob :- s/Str nil}
+                                    {email :- s/Str nil}
+                                    {name :- s/Str nil}
+                                    {personOid :- s/Str nil}
+                                    {applicationOid :- s/Str nil}]
+                     :summary "Return applications header-level info for form"
+                     :return {:applications [ataru-schema/ApplicationInfo]}
+                     (let [[query-key query-value] (cond
```

Comment:
Could probably use `(condp some?`

Reply:
AFAIK `condp` doesn't really work here since it requires both a predicate and an expression..